project-root/Assets/Scripts/NavMeshSpawner.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class NavMeshSpawner : MonoBehaviour
+{
+    public GameObject monsterSpawn;
+    Vector3 point;
+    public float time;
+    public bool done;
+    public List<GameObject> monsters = new List<GameObject>();
+    //public int enemyIndex;
+    public int spawned;
+    public int amountToSpawn;
+    public int reSpawnTime;
+
+    public float range = 10f;
+
+    bool RandomPoint(Vector3 center, float range, out Vector3 result)
+    {
+        
+        for (int i = 0; i < 10; i++)
+        {
+            
+            Vector3 randomPoint = center + Random.insideUnitSphere * range;
+            NavMeshHit hit;
+            if (NavMesh.SamplePosition(randomPoint, out hit, 1 , NavMesh.AllAreas))

```

Comment:
You don't want it to check AllAreas, because it will return true even if you're hitting a "Not Walkable" navmesh, i.e. a table or chair.

Reply:
it wont spawn on an not walk-able though

Reply:
NavMesh.AllAreas is stating that all Navmesh area types are accepted though. Where else are you explicitly stating that you don't want objects to spawn on "not walkable"?

Reply:
NavMesh.SamplePosition finds the closes point on the navmesh which is the  walk able area. it counts as an obstacle if its not walk-able. I've tested it to and not once has it spawned on a non-walk-able object only around on the blue walk-able areas. This function I believe is fine and it works well.

Reply:
Yes but NavMesh.SamplePosition is expensive in itself, so to help shave down on the workload of the function only pass through NavMesh.Walkable, that way it's only searching on walkable areas not on every navmesh in the level.